warehouse/templates/manage/release.html

```diff
@@ -126,7 +126,9 @@ <h3>Delete release</h3>
         files
       {% endtrans %}
     {% else %}
-      Deleting will irreversibly delete this release
+      Deleting will irreversibly delete this release.
+      You will not be able to re-upload a new distribution of the same type with the same version number.
+      Consider making a new release or a post release instead
```

Comment:
Can we link to https://www.python.org/dev/peps/pep-0440/#post-releases on "post release"?